cluster/juju/layers/kubernetes-master/README.md

```diff
@@ -1,118 +1,102 @@
+[getting started guide](http://kubernetes.io/docs/user-guide/#concept-guide).
 
-This charm was built from other charm layers using the reactive framework. The
-`layer:docker` is the base layer. For more information please read [Getting
-Started Developing charms](https://jujucharms.com/docs/devel/developer-getting-started)
+This charm is an encapsulation of the Kubernetes master processes and the 
+operations to run on any cloud for the entire lifecycle of the cluster.
 
-# Deployment
-The kubernetes charms require a relation to a distributed key value store
-(ETCD) which Kubernetes uses for persistent storage of all of its REST API
-objects.
-
-```
-juju deploy trusty/etcd
-juju deploy kubernetes-master
-juju deploy kubernetes-node
-juju add-relation kubernetes-master etcd
-juju add-relation kubernetes-node etcd
-```
+This charm was built from other charm layers using the Juju reactive framework.
+The other layers focus on specific subset of operations making this layer 
+specific to operations of Kubernetes master processes.
 
-# Configuration
-For your convenience this charm supports some configuration options to set up
-a Kubernetes cluster that works in your environment:  
+This charm is one part of what is needed to model a Kubernetes cluster. If you
+wish to deploy a complete cluster please look at the collection of charms that
+model a [Kubernetes core](https://jujucharms.com/kubernetes-core/) cluster.
 
-#### dns_domain
-
-The domain name to use for the Kubernetes cluster by the skydns service.
-
-#### enable-dashboard-addons
-
-Enables the installation of Kubernetes dashboard, Heapster, Grafana, and
-InfluxDB.
+# Deployment
 
-# Additional Information
+This charm is not self contained, other charms are needed to model a complete 
+Kubernetes cluster. The cluster requires a distributed key value store such as 
+[Etcd](https://coreos.com/etcd/) and the kubernetes-worker charm which delivers
+the Kubernetes node services. A cluster requires a Software Defined Network 
+(SDN) and Transport Layer Security (TLS) so the components in a cluster 
+communicate securely. 
 
-## Kubedns
+Please take a look at the [Canonical Distribution of Kubernetes](https://jujucharms.com/canonical-kubernetes/) 
+or the [Kubernetes core](https://jujucharms.com/kubernetes-core/) bundles for 
+examples of complete models of Kubernetes clusters.
 
-Kubernetes DNS is handled via the kubedns addon. More information about
-this service can be obtained in the
-[Kubernetes DNS admin guide](http://kubernetes.io/docs/admin/dns/ )
+# Resources
 
+The kubernetes-master charm takes advantage of the [Juju Resources](https://jujucharms.com/docs/2.0/developer-resources) 
+feature.
 
-# Storage
-The kubernetes charm is built to handle multiple storage devices if the cloud
-provider works with
-[Juju storage](https://jujucharms.com/docs/devel/charms-storage).
+The resources are already attached to the Juju Charm Store, but in cases where
```

Comment:
We need to be fairly careful about this messaging. We dont give end users any information on how to package their own release of Kubernetes. Perhaps we change the byline to this:

```
Kubernetes releases being delivered in the charms are evaluated by the Canonical ~containers team. They are provided for you by the charms in the Charm Store. If you are deploying juju in a network limited cloud, you will be responsible for creating these resources, or fetching them before deployment.

... insert additional info here if applicable ...

```

I dont know thats any better than whats here, but just a suggestion.